django_airavata/apps/api/serializers.py

```diff
@@ -513,6 +515,13 @@ def get_isInputFileUpload(self, data_product):
         request = self.context['request']
         return user_storage.is_input_file(request, data_product)
 
+    def get_path(self, data_product):
+        """Getter for path field."""
+        if len(data_product.replicaLocations) > 0:
+            return re.sub(r'.*/tmp/[^/]*/', "", data_product.replicaLocations[0].filePath)
```

Comment:
This won't work because not all locations are in /tmp/. That's just kind of a local development setup. I think we just need to treat data products differently than paths in user storage.

Reply:
@machristie Do you have any thoughts on how to do that. Currently, I cannot see any existing utilities or any other way.

Reply:
Yeah, see my other comment below. Essentially, if we have a data product URI, that should be all that the API needs. The API shouldn't need the path because the path can be determined from the data product URI. I'll create an updated `user_storage.update_file_content` function in the SDK that you can use in DataProductView.put()

Reply:
@dinukadesilva I've added an update_data_product_content function to the user_storage.py module - https://github.com/apache/airavata-django-portal-sdk/blob/master/airavata_django_portal_sdk/user_storage.py#L268.

To use the latest sdk, you'll need to clone it and then install it into your airavata-django-portal virtual environment
```
# activate your airavata-django-portal venv
cd airavata-django-portal
source venv/bin/activate
cd ..

# clone and install SDK into portal venv
git clone https://github.com/apache/airavata-django-portal-sdk.git
cd airavata-django-portal-sdk
python setup.py develop
```

Reply:
This worked @machristie . Thanks